viewing_party/party.py

```diff
@@ -1,23 +1,270 @@
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def get_unique_watched(user_data):
+    '''
+    collects uniqe movies watched by user compared to friends and
+    stores it in a list, then returns that list
+    '''
+    unique_watched_list = []
+
+    friend_watched_set = set(get_friends_watched(user_data))
+    user_watched_set = set(get_users_watched(user_data))
+
+    unique_set = user_watched_set - friend_watched_set
+
+    for i in range(len(user_data["watched"])):
+        if user_data["watched"][i]['title'] in unique_set:
+            unique_watched_list.append(user_data["watched"][i])   
+
+    return unique_watched_list
+    
+    
+    
+def get_friends_watched(user_data):
+    '''
+    Helper Function - Stores the titles of the friends 'watched' list from 
+    user_data in a list, and returns that list
+    '''   
+    friends_list_of_dict = user_data["friends"]
+    friends_watched_list = []
+
+    for i in range(len(friends_list_of_dict)):
+        for watched in friends_list_of_dict[i]:
+            for movie in friends_list_of_dict[i][watched]:
+                friends_watched_list.append(movie['title'])
+
+    return friends_watched_list
+
+def get_users_watched(user_data):
+    '''
+    Helper Function - Stores the titles of the users 'watched' list from 
+    user_data in a list, and returns that list
+    '''      
+    
+    user_watched_list = user_data["watched"]
+    users_watched_movies = []
+
+    for movie in user_watched_list:
+        users_watched_movies.append(movie['title'])
+
+    return users_watched_movies
+
+
+def get_friends_unique_watched(user_data):
+    '''
+    collects unique watched movies from friends, stores them in a list,
+    and returns the list
+    '''
+    unique_watched_list = []
+
+    # User Data Watched Sets
+    friend_watched_set = set(get_friends_watched(user_data))
+    user_watched_set = set(get_users_watched(user_data))
+
+    unique_set = friend_watched_set - user_watched_set
+
+    # User Data Variables
+    friends_list_dict = user_data["friends"]
+
+
+    for i in range(len(friends_list_dict)):
+        for friend_movie in friends_list_dict[i]["watched"]:
+            if friend_movie["title"] in unique_set and friend_movie not in unique_watched_list:
+                unique_watched_list.append(friend_movie) 
+
+    return unique_watched_list
+
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
+def get_available_recs(user_data):
+    '''
+    Returns a list of recommended movies based on what
+    the user has a subscription to using user_data as input
+    '''
+    recommended_movies_list = []
+
+    # User Data Watched Sets
+    friend_watched_set = set(get_friends_watched(user_data))
+    user_watched_set = set(get_users_watched(user_data))
+    
+    unique_set = friend_watched_set - user_watched_set

```

Comment:
This function is doing a lot of the same work as `get_friends_unique_watched`.  Could you either use that function to simplify the code here, or pull some of the shared code out into a helper function?